components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/GeckoCreditCardsAddressesStorageDelegate.kt

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.storage.sync
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import mozilla.components.concept.storage.Address
+import mozilla.components.concept.storage.CreditCard
+import mozilla.components.concept.storage.CreditCardsAddressesStorage
+import mozilla.components.concept.storage.CreditCardsAddressesStorageDelegate
+
+/**
+ * [CreditCardsAddressesStorageDelegate] implementation.
+ */
+class GeckoCreditCardsAddressesStorageDelegate(
+    private val storage: Lazy<CreditCardsAddressesStorage>,
+    private val scope: CoroutineScope = CoroutineScope(Dispatchers.IO)
+) : CreditCardsAddressesStorageDelegate {
+
+    override fun onAddressFetch(): Deferred<List<Address>> {
+        return scope.async {
+            storage.value.getAllAddresses()
+        }
+    }
+
+    override fun onAddressSave(address: Address) {
+        TODO("Not yet implemented")
```

Comment:
I think `onAddressSave` and `onCreditCardSave` depends on our validation delegates which I have filed separately in https://github.com/mozilla-mobile/android-components/issues/9803

Reply:
Just filed https://github.com/mozilla/application-services/issues/3899 - with that API in place, these calls can just go directly to storage.

E.g. at this point an address (and a credit card below) have been validated as part of the prompt flow, and we're just being asked by GV to save the data. Logins does validation here in order to figure out if it's an update or a save, but hopefully we won't need to do that.